Sources/SwiftDiagnostics/DiagnosticsFormatter.swift

```diff
@@ -103,12 +103,76 @@ public struct DiagnosticsFormatter {
         }
 
         for diag in diags.dropLast(1) {
-          annotatedSource.append("\(preMessage)├─ \(diag.message)\n")
+          if colorize {
+            annotatedSource.append("\(preMessage)├─ \(self.colorize(diag.diagMessage))\n")
+          } else {
+            annotatedSource.append("\(preMessage)├─ \(diag.message)\n")
+          }
         }
-        annotatedSource.append("\(preMessage)╰─ \(diags.last!.message)\n")
 
+        if colorize {
+          annotatedSource.append("\(preMessage)╰─ \(self.colorize(diags.last!.diagMessage))\n")
+        } else {
+          annotatedSource.append("\(preMessage)╰─ \(diags.last!.message)\n")
+        }
       }
     }
     return annotatedSource
   }
+
+  /// Annotates the given ``DiagnosticMessage`` with an appropriate ANSI color code and returns the result as a printable string.
+  static func colorize(_ message: DiagnosticMessage) -> String {
+    switch message.severity {
+    case .error:
+      let annotation = ANSIAnnotation(color: .red, trait: .bold)
+      return annotation.applied(to: message.message)
+    case .warning:
+      let annotation = ANSIAnnotation(color: .yellow)
+      return annotation.applied(to: message.message)
+    case .note:
+      return message.message
+    }
+  }
+
+  enum ANSIColor: UInt8 {
+    case reset = 0
+    case black = 30
+    case red = 31
+    case green = 32
+    case yellow = 33
+    case blue = 34
+    case magenta = 35
+    case cyan = 36
+    case white = 37
+  }
+
+  enum ANSITrait: UInt8 {
+    case bold = 1
```

Comment:
If you’ve got the `reset` color, I think you should also have `ANSITrait.reset`. Or alternatively remove `ANSIColor.reset` and use nil to represent resetting the color. I think I slightly prefer using `nil` but don’t really have strong opinions. It would just be good to have this be consistent.

Reply:
Agree on the consistency note. What about introducing the reset (I've noticed the alternative naming `normal`) case on `ANSITrait` and having that as the default? Maybe it would make more sense to have that and `ANSIColor.reset` just be called `normal` in that case though.

Reply:
Yes, I think having `ANSITrait.reset` would be equally good. And `normal` sounds better to me as well. Makes it sound less like a state machine that you need to reset.